File: 374-guess-number-higher-or-lower/guess-number-higher-or-lower.py
class Solution:
    def guessNumber(self, n: int) -> int:
        first = 0
        last = n - 1

        while(first <= last):
            mid = (first + last) // 2
            ans = guess(mid)

            if(ans == 0):
                return mid
            elif(ans < 0):
                last = (mid - 1)
            else:
                first = (mid + 1)

        return first

File: 374-guess-number-higher-or-lower/guess-number-higher-or-lower.js
let guessNumber = function(n)
{
    let first = 1;
    let last = n;
    while (first <= last)
    {
        let mid = Math.ceil(first + (last - first) / 2);
        let pick = guess(mid)

        if (pick == 0)
        {
            return mid;
        }
        else if (pick < 0)
        {
            last = mid - 1;
        } 
        else
        {
            first = mid + 1;
        }
    }
    
    return -1;
};